instrumentation/src/main/java/io/opentelemetry/android/internal/services/PreferencesService.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.internal.services;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import io.opentelemetry.android.BuildConfig;
+
+/**
+ * Utility to store and retrieve apps' preferences.
+ *
+ * <p>This class is internal and not for public use. Its APIs are unstable and can change at any
+ * time.
+ */
+public class PreferencesService implements Service {
+    private final SharedPreferences preferences;
+
+    public PreferencesService(Context context) {
+        preferences =
+                context.getSharedPreferences(
+                        BuildConfig.LIBRARY_PACKAGE_NAME + ".prefs", Context.MODE_PRIVATE);
+    }
```

Comment:
Code smell -- wrong dependency passed to constructor. I'd make this a factory method `create(context)` and pass the `SharedPreferences` instance into a new constructor.

Reply:
Ideally this is also lazily loaded (or off of the main thread) instead of in the ctor since this does IO and will be warned by `StrictMode`.

Reply:
> I'm assuming this will be created once the SDK starts on the main thread

You're right. The idea is to initialize it in `OpenTelemetryRumBuilder` which should be called in the App.onCreate method. We're going to have to take a look at what parts of the initialization need to happen in a worker thread.

Reply:
Good call @marandaneto. Would you mind opening an issue to track this so that we don't have to force it into this sizable PR? Thanks.

Reply:
Ok, I've went ahead and logged #149 for this.